Eastern-Research-Group/expert-query
# Feature/145 client side parameter error

## Related Issues:
* [EQ-144](https://jira.epa.gov/secure/RapidBoard.jspa?rapidView=865&view=detail&selectedIssue=EQ-144)
* [EQ-145](https://jira.epa.gov/secure/RapidBoard.jspa?rapidView=865&view=detail&selectedIssue=EQ-145)

## Main Changes:
* Instead of silently discarding a URL parameter, the user interface displays a message describing the issues discovered
    * This message lists all discovered instances of invalid or duplicate (when multiple instances not allowed) parameters.
* On the server side, the same checks are performed, but an error response is sent as soon as a bad parameter is encountered.
    * This is done because unexpected results would be returned if the parameter was dropped. In the case of the UI, the parameter parsing is only done to populate input fields before performing an actual query.

## Steps To Test:
1. Navigate to http://localhost:3000/attains/catchmentCorrespondence#format=csv&reportingCycleLo=2001&reportingCycleLo=2003&actionNickname=Tim, and confirm a message box is show that lists `reportingCycleLo` as a duplicate parameter and `actionNickname` as an invalid parameter. The `from` field of the "Reporting Cycle" input set should be populated with the value of the first encountered low parameter, 2001.
2. Navigate to http://localhost:3000/attains/assessmentUnits#actionId=10715 and confirm a message box is show with `actionId` listed as an invalid parameter for the selected profile ("Assessment Units").
3. Navigate to http://localhost:9090/attains/data/actions?format=csv&actionNickname=Tim and confirm a JSON message is returned with `actionNickname` as an invalid parameter.
4. Navigate to http://localhost:9090/attains/data/catchmentCorrespondence?format=tsv&reportingCycleLo=3000&reportingCycleLo=2001 and confirm a JSON message is returned with `reportingCycleLo` as a duplicate parameter.
5. Navigate to http://localhost:3000/attains/catchmentCorrespondence#format=csv&assessmentUnitName=Abalone%20Cove%20Beach&assessmentUnitName=Abalone%20Cove%20Beach and confirm that "Abalone Cove Beach" is properly decoded from the URL string, and that only one instance of the value is used to populate the "Assessment Unit Name" input field.
